### ISSUE TITLE
Add an Example for DeepNeuralNetworkLearning

### ISSUE THREAD

**Issue by FireDragonGameStudio** [accord-net/framework#1013]
**What would you like to submit?** (put an 'x' inside the bracket that applies)
  - [x] question
  - [ ] bug report
  - [ ] feature request

**Issue description**
Is it possible to add an example for the usage of DeepNeuralNetworkingLearning? I already tried a lot, but can't get it working.

**Comment by blaisexen** [comment]
hi,
you can find samples under this folder,
https://github.com/accord-net/framework/tree/development/Samples/Neuro

if specifically for image classification, here
https://github.com/accord-net/framework/tree/development/Samples/Neuro/Classification%20(ANN)

Good Luck

**Comment by FireDragonGameStudio** [comment]
Thx for pointing me in the right direction :) But when I execute a simple for XOR with the code in the deep learning sample, the network is not learning at all (or showing wrong results). I'm using Accord.NET 3.8 in Unity 2017.1.1 btw. I'll attach you my sample code.
```C#
double[][] inputs = new double[4][] {
    new double[] {0, 0}, new double[] {0, 1},
    new double[] {1, 0}, new double[] {1, 1}
};
double[][] outputs = new double[4][] {
    new double[] {0}, new double[] {1},
    new double[] {1}, new double[] {0}
};

// create network
DeepBeliefNetwork neuralNetwork = new DeepBeliefNetwork(new BernoulliFunction(), 2, 2, 1);

// generate weights
new GaussianWeights(neuralNetwork).Randomize();
neuralNetwork.UpdateVisibleWeights();

DeepNeuralNetworkLearning neuralTeacher = new DeepNeuralNetworkLearning(neuralNetwork) {
    Algorithm = (ann, i) => new ParallelResilientBackpropagationLearning(ann),
    LayerIndex = neuralNetwork.Layers.Length - 1
};

double[][] layerData = neuralTeacher.GetLayerInput(inputs);

for (int i = 0; i < 5000; i++) {
    double error = neuralTeacher.RunEpoch(layerData, outputs);
    if (i % 100 == 0) {
        Debug.Log("Supervised -> " + i + ", Error = " + error);
    }
}
neuralNetwork.UpdateVisibleWeights();

int correct = 0;
for (int i = 0; i < inputs.Length; i++) {
    double[] outputValues = neuralNetwork.Compute(inputs[i]);
    double outputResult = outputValues.First() >= 0.5 ? 1 : 0;

    Debug.Log("For inputs " + String.Join(",", inputs[i].Select(o => o.ToString()).ToList().ToArray()) + " output " + outputResult + " was learned");

    if (outputResult == outputs[i].First()) {
        correct++;
    }
}

Debug.Log("Correct " + correct + "/" + inputs.Length + ", " + Mathf.Round(((float)correct / (float)inputs.Length * 100)) + "%");

**Comment by blaisexen** [comment]
@FireDragonGameStudio maybe we have to wait for @cesarsouza

**Comment by FireDragonGameStudio** [comment]
Hy,
as I was playing around with this sample, I found a way to get it up and running, with the help of ActivationNetwork and ParallelResilientBackpropagationLearning. I guess this will be a sufficient workaround for DeepNeuralNetworkLearning, but the main (and more obviously) reason is an usage error from my side, as the DeepNeuralNetworkLearning class is maybe not supposed to be used that way I intended. Please find my version attached :)

```C#
double[][] inputs = new double[4][] {
    new double[] {0, 0}, new double[] {0, 1},
    new double[] {1, 0}, new double[] {1, 1}
};
double[][] outputs = new double[4][] {
    new double[] {0}, new double[] {1},
    new double[] {1}, new double[] {0}
};

double initialStep = 0.0125;
double sigmoidAlphaValue = 2.0;

// create neural network
ActivationNetwork neuralNetwork = new ActivationNetwork(new SigmoidFunction(sigmoidAlphaValue), 2, 2, 1);

// create teacher
ParallelResilientBackpropagationLearning neuralTeacher = new ParallelResilientBackpropagationLearning(neuralNetwork);

// set learning rate and momentum
neuralTeacher.Reset(initialStep);

for (int i = 0; i < 5000; i++) {
    double error = neuralTeacher.RunEpoch(inputs, outputs);
    if (i % 100 == 0) {
        Debug.Log("Supervised -> " + i + ", Error = " + error);
    }
}

int correct = 0;
for (int i = 0; i < inputs.Length; i++) {
    double[] outputValues = neuralNetwork.Compute(inputs[i]);
    double outputResult = outputValues.First() >= 0.5 ? 1 : 0;

    Debug.Log("For inputs " + String.Join(",", inputs[i].Select(o => o.ToString()).ToList().ToArray()) + " output " + outputResult + " was learned");

    if (outputResult == outputs[i].First()) {
        correct++;
    }
}

Debug.Log("Correct " + correct + "/" + inputs.Length + ", " + Mathf.Round(((float)correct / (float)inputs.Length * 100)) + "%");
```

**Comment by XYC-EXP** [comment]
Thank you very much.
